src/manager.spec.js

```diff
@@ -1,173 +1,845 @@
-describe('AdsManager', function () {
-
-  var ads;
+describe('AdManager', function() {
+  var AdManager, AdManagerWrapper, adManager;
+  var MockGoogleTag = require('mock_google_tag');
   var adUnits = require('bulbs.ads.units');
-  var MockGoogleTag = require('mockGoogleTag');
-  var sandbox = sinon.sandbox.create();
 
-  beforeEach(function () {
+  beforeEach(function() {
+    AdManagerWrapper = require('./manager');
     window.googletag = new MockGoogleTag();
 
-    ads = require('./manager.js');
-    ads.init();
+    window.TARGETING = {
+      dfp_site: 'onion',
+      dfp_pagetype: 'homepage'
+    };
+    adManager = AdManagerWrapper.init();
   });
 
-  afterEach(function () {
-    sandbox.restore();
-    var added = document.querySelectorAll('body > div');
+  describe('new AdManager', function() {
+    it('has no slots yet', function() {
+      expect(adManager.slots).to.eql({});
+    });
 
-    for (var i = 0, l = added.length; i < l; i ++) {
-      var v = added[i];
-      document.body.removeChild(v);
-    }
+    it('has a default ad id', function() {
+      expect(adManager.adId).to.equal(0);
+    });
+
+    it('merges global TARGETING with ad unit dfp site param', function() {
+      expect(adManager.targeting).to.eql({
+        dfp_site: 'onion',
+        dfp_pagetype: 'homepage'
+      });
+    });
+
+    it('is not initialized', function() {
+      expect(adManager.initialized).to.be.false;
+    });
+
+    xit('binds context for all event handlers', function() {
```

Comment:
Could split the wrapper and the manager into two modules:

#### `ads-manager.js`
```js
module.exports = AdsManager;
```

#### `ads-manager-init.js`
```js
var AdsManager = require('./ads-manager');
module.exports.init = function (options) {
  return new AdsManager(options);
}
```

Files could be named whatever, but then the `init` file is the main public module. Then your test can import the class directly.